### PR TITLE
feat: add array buffer and regexp to live objects, no longer memoize toStringTags

### PR DESCRIPTION
**Pull request by jdalton**
feat: add array buffer and regexp (because of .lastIndex) to live objects, no longer memoize toStringTags, and add lots of tests.

### CODE REVIEW

File: test/membrane/life-red-proxies.spec.js [hunk 1]
```diff
@@ -8,77 +8,383 @@ class ExoticObject {
...
+const regexp = /a/;
+const uint8ClampedArray = new Uint8ClampedArray(buffer);
+const uint16Array = new Uint16Array(buffer);
+const uint32Array = new Uint32Array(buffer);
+
 const plainObject = {
     x: 'uno',
 };
 
+const planObjectNullProto = {
```
Comment (rwaldron) [inline test/membrane/life-red-proxies.spec.js:29]:
```suggestion
const plainObjectNullProto = {
```

File: test/membrane/life-red-proxies.spec.js [hunk 2]
```diff
@@ -8,77 +8,383 @@ class ExoticObject {
...
         exoticLiveObject,
         expect,
+        float32Array,
+        float64Array,
+        int8Array,
+        int16Array,
+        int32Array,
         jasmine,
         plainObject,
+        planObjectNullProto,
```
Comment (rwaldron) [inline test/membrane/life-red-proxies.spec.js:51]:
```suggestion
        plainObjectNullProto,
```

File: test/membrane/life-red-proxies.spec.js (hunk 1 above)
Comment (jdalton) [reply test/membrane/life-red-proxies.spec.js:29]: That naming did not go according to plain.

File: test/membrane/life-red-proxies.spec.js [hunk 3]
```diff
@@ -8,77 +8,383 @@ class ExoticObject {
...
+const float32Array = new Float32Array(buffer);
+const float64Array = new Float64Array(buffer);
+const int8Array = new Int8Array(buffer);
+const int16Array = new Int16Array(buffer);
+const int32Array = new Int32Array(buffer);
+const uint8Array = new Uint8Array(buffer);
+const regexp = /a/;
+const uint8ClampedArray = new Uint8ClampedArray(buffer);
+const uint16Array = new Uint16Array(buffer);
+const uint32Array = new Uint32Array(buffer);
```
[the remark this answers is not in the document]
Comment (jdalton) [reply test/membrane/life-red-proxies.spec.js:23]: Yes

[the remark this answers is not in the document]
Comment (jdalton) [reply test/membrane/life-red-proxies.spec.js:25]: Added

File: test/membrane/life-red-proxies.spec.js (hunk 2 above)
Comment (jdalton) [reply test/membrane/life-red-proxies.spec.js]: Fixed

**Review by rwaldron — APPROVED** [review]

**Pull request merged**